src/main/java/edu/harvard/h2ms/service/UserServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package edu.harvard.h2ms.service;
+
+import com.google.common.collect.Lists;
+import edu.harvard.h2ms.domain.core.Event;
+import edu.harvard.h2ms.repository.EventRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.springframework.stereotype.Repository;
+import org.springframework.stereotype.Service;
+import edu.harvard.h2ms.domain.core.User;
+import edu.harvard.h2ms.repository.UserRepository;
+import org.springframework.transaction.annotation.Transactional;
+import static edu.harvard.h2ms.service.utils.H2msRestUtils.calculateAverage;
+import static java.lang.Boolean.TRUE;
+
+@Service("userService")
+@Repository
+@Transactional
+public class UserServiceImpl implements UserService {
+
+	final Logger log = LoggerFactory.getLogger(UserServiceImpl.class);
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private EventRepository eventRepository;
+
+	@Override
+	@Transactional(readOnly=true)
+	public Map<String, Double> findAvgHandWashCompliance() {

```

Comment:
I think we should have a generic way to think about this based on an EventTemplate.

Reply:
Open another jira for refactoring with details. This api should still be usable in the iterim

Reply:
How would this be usable? It’s not using the data model. Each event has Answers associated with it, that’s where the value for Washed is.

Reply:
I think some generic approach would mean finding all the events with an associated answer to a particular question and returning the percentage for each of answer values (e.g. percent true and false for Washed). 

For hand washing compliance there’s two questions involved (washed and opportunity) although I’m not clear what the exact formula is for compliance.